bokeh/models/axes.py

```diff
@@ -124,10 +124,22 @@ class Axis(GuideRenderer):
     of ticks.
     """)
 
-    axis_label = Nullable(Either(String, Instance(MathText)), default="", help="""
+    _axis_label = Nullable(Either(String, Instance(MathText)), default="", help="""
     A text or LaTeX notation label for the axis, displayed parallel to the axis rule.
     """)
 
+    @property
+    def axis_label(self) -> Nullable(Either(String, Instance(MathText))):
+        return self._axis_label
+
+    @axis_label.setter
+    def axis_label(self, value: Either(String, Instance(MathText))) -> None:
+        if (isinstance(value, str)):
+            if(value.startswith("$") and value.endswith("$")):
+                self._axis_label = MathText(text=value)
+        else:
+            self._axis_label = value
```

Comment:
This is not the right way to go about this, for a few reasons. We should use a `.accepts` clause on this. Ideally we could define a new string/regex property `MathString` type that matches `$......$` strings, then 
```
axis_label = Nullable(Either(String, Instance(MathText)), ...).accepts(MathString, lambda val: MathText(text=val))
```

Reply:
a `MathString` might look like:
```
class MathString(String):
    def validate(self, value, detail=True):
        super().validate(value, detail)

        if  len(value) > = 2 and value[0] == value[-1] == "$":
            return

        msg = "" if not detail else f"expected a math string"
        raise ValueError(msg)
```

Reply:
> axis_label = Nullable(...).accepts(MathString, lambda val: MathText(text=val[1:-1]))

wouldn't this imply that we only accept math strings? we agreed on accepting all of [these formats](https://github.com/bokeh/bokeh/discussions/11395#discussioncomment-953157)



I will add the `PlainText` model to allow strings without interpretation.

Reply:
@IuryPiva No, the `accepts` conversion function is only used if the value type-validates against the `accepts` type.  In this case `MathText` will only validate strings like `$.....$` I think we may not need `PlainText` it's possible the existing `String` will suffice to mark plain text

Reply:
Alright, thanks Bryan

Reply:
Can you explain what corner cases are you thinking of when you mention that `$` might be in the center of the string?

Reply:
I just mean it's always good to consider explicit what can, should, and does happen in corner cases. In this situation, things like `"$...$..$"` or `"$...\$....$" or `"$$...$"` etc.  We should make sure there are tests that cover cases like this *now*, so that if the implementation ever changes in the future, they will already be covered.